src/plugins/query_enhancements/public/query_assist/components/query_assist_summary.tsx

```diff
@@ -206,6 +209,21 @@ export const QueryAssistSummary: React.FC<QueryAssistSummaryProps> = (props) =>
     });
   }, [props.core]);
 
+  useEffect(() => {
+    setIsSummaryAgent(false);
+    const fetchSummaryAgent = async () => {
+      if (selectedDataset.current?.dataSource?.id) {
+        const summaryAgentStatus = await checkAgentsExist(
```

Comment:
What will happen when `checkAgentsExist` throw exception? Should we try/catch the exception?